### PR TITLE
Fix compile issues with Next.js 13 using the app directory

### PR DESCRIPTION
**Pull request by chrissiwaffler**
Fixes the problem described in https://github.com/MeshJS/mesh/issues/124

by applying the solution described by https://github.com/Emurgo/cardano-serialization-lib/issues/157#issuecomment-1396933301.

---

This means switching the dependency of the cardano libs to `asmjs`.

---

I tested it with my Next.js 13 project using the new `app`-directory-structure and it works!

### CODE REVIEW

**Review by abdelkrimdev — APPROVED** [review]

**Comment by markrufino** [comment]
Hi @chrissiwaffler  @abdelkrimdev sorry to bother you guys, I've been researching for almost half a day now. How should I add MeshJS in my `package.json` if I want to install `core` and `react` from the `next13` branch instead of the release? Thanks the usual examples don't work. I'm encountering this issue as well because I'm using Next13 App Router.

**Comment by chrissiwaffler** [comment]
Hi @markrufino,

## TL;DR: 

change the dependencies in the `package.json` file of your Next13 Project to the following:
```
  "dependencies": {
    "@meshsdk/core": "github:stargazerlabs/mesh-core",
    "@meshsdk/react": "^1.1.7",
  }
```
---

## Explanation

I had the same problem and solved it by extracting the `module` folder of this mesh mono-repository into an own git repository.
This way, the `package.json` of the `module` folder is now at the root and is accessible via npm.
It is also important to know, that the `module` folder represents the `@meshsdk/core` as given by the `name` attribute of the `package.json` in there.

 Then, I executed the **build** command of the module folder and added the generated files and the rest of the code to a self-hosted git.
The **build** step is needed, so you can import the `@meshsdk/core` later into your project.

This git repo is publicly accessible via https://github.com/StargazerLabs/mesh-core.
Now you can add this line under `"dependencies"` in your `package.json`: 

```
"@meshsdk/core": "github:stargazerlabs/mesh-core",
```

---

Feel free to write again, if you have any questions about that!

**Comment by markrufino** [comment]
@chrissiwaffler ,

Thank you very much! I was able to get it working with that solution, although I had to use `--force` for installing `@meshsdk/react` after because I'm encountering dependency issues with it despite using the version you've indicated.

**Comment by nguyenvuhoang** [comment]
I tried using --force for import @meshsdk/core": "github:stargazerlabs/mesh-core. but npm still got error message can't install

**Comment by chrissiwaffler** [comment]
what's the full error message? @nguyenvuhoang

**Comment by chrissiwaffler** [comment]
Currently I'm also using our own forked react module: 
```
    "@meshsdk/core": "github:stargazerlabs/mesh-core",
    "@meshsdk/react": "github:stargazerlabs/mesh-react",
```

and everything is under `"dependencies"` in my `package.json` file

**Comment by nguyenvuhoang** [comment]
Hehe. I changed "@meshsdk/react": "github:stargazerlabs/mesh-react". it worked right now. 
because i see you mentioned that using @meshsdk/react@1.1.7

Thank very much!!!!!!!!!!

**Comment by chrissiwaffler** [comment]
Ah alright, good to know!

Love to see it working for you! :)

**Comment by olliejm** [comment]
Is there anything blocking this change from being released in the npm package?
